packages/functions/src/dal/migrationScripts/ChangeAppointmentStatusMigration.ts

```diff
@@ -0,0 +1,39 @@
+import { getAllLatestAppointments, setAppointment } from "../AppointmentDataAccessLayer";
+import { AppointmentStatus } from "@zm-blood-components/common/src";
+const fs = require('fs');
+
+export async function addStatusForAppointments(){
+  const appointments = await getAllLatestAppointments();
+
+  return appointments.map((appointment) => {
+    let newStatus = AppointmentStatus.AVAILABLE;
+    const now = new Date();
+    if(appointment.donorId !== "")
+    {
+      if (appointment.donationStartTime.toDate() < now)
+        newStatus = AppointmentStatus.COMPLETED; // should this be completed?
+      else
+        newStatus = AppointmentStatus.BOOKED;
+    }
+    else
+    {
+      if(appointment.donationStartTime.toDate() < now)
+        newStatus = AppointmentStatus.NOSHOW
+    }
+    return {
+      status: newStatus,
+      ...appointment
+    };
+  });
+}
+
+export async function migrateAppointmentStatus(){
+  const updatedAppointments = await addStatusForAppointments();
+  const stream = fs.createWriteStream('results.txt', {flags:'a'});
```

Comment:
Where is this file saved? 
Why is this needed? 
Can logs be enough?

Reply:
didn't see we used log anywhere... I guess it would work...I just wanted a single file to give me the result...it would be saved in the folder the script would run from....

Reply:
See log line example here:
https://github.com/startach/ZM-BloodComponents/blob/2f5fa1ebe6a2e83d1ab434c8cf8eaed4db0e05c4/packages/functions/src/RequestHandleWrapper.ts#L18